presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/TransformCorrelatedSingleRowSubqueryToProject.java

```diff
@@ -72,13 +72,28 @@ public Result apply(LateralJoinNode parent, Captures captures, Context context)
                 .where(ProjectNode.class::isInstance)
                 .findAll();
 
-        if (subqueryProjections.size() == 0) {
+        // LateralJoinNode's input's output symbols will be added as identity projections
+        // during building the result ProjectNode. However, InlineProjections might have
+        // added the same identity projection, so only keep one of them
```

Comment:
but then `RemoveRedundantIdentityProjections` should remove this projection. Please undo these changes.

Reply:
No, we need this code block because of two reasons:

1) ``RemoveRedundantIdentityProjections`` will remove projections only when both the Projection's assignments and its's source assignments are same. In this case, the Projection's source will have no assignments.

2) The projection being removed in this code block will be added back when this rule constructs its result. So this is technically an idempotent operation. However, this code block is needed to allow ``TransformCorrelatedSingleRowSubqueryToProject`` to fire and actually build the result.

I will re word the comment since we use the term "identity projections" for something slightly different.